include/faabric/transport/MessageEndpointServer.h

```diff
@@ -1,72 +1,49 @@
 #pragma once
 
+#include <faabric/proto/faabric.pb.h>
 #include <faabric/transport/Message.h>
-#include <faabric/transport/MessageContext.h>
 #include <faabric/transport/MessageEndpoint.h>
-#include <faabric/transport/MessageEndpointClient.h>
+#include <faabric/util/latch.h>
 
 #include <thread>
 
-#define ENDPOINT_SERVER_SHUTDOWN -1
-
 namespace faabric::transport {
-/* Server handling a long-running 0MQ socket
- *
- * This abstract class implements a server-like loop functionality and will
- * always run in the background. Note that message endpoints (i.e. 0MQ sockets)
- * are _not_ thread safe, must be open-ed and close-ed from the _same_ thread,
- * and thus should preferably live in the thread's local address space.
- */
+
+// This server has two underlying sockets, one for synchronous communication and
+// one for asynchronous.
 class MessageEndpointServer
 {
   public:
-    MessageEndpointServer(int portIn);
+    MessageEndpointServer(int asyncPortIn, int syncPortIn);
 
-    /* Start and stop the server
-     *
-     * Generic methods to start and stop a message endpoint server. They take
-     * a, thread-safe, 0MQ context as an argument. The stop method will block
-     * until _all_ sockets within the context have been closed. Sockets blocking
-     * on a `recv` will be interrupted with ETERM upon context closure.
-     */
-    void start(faabric::transport::MessageContext& context);
+    virtual void start();
 
-    void stop(faabric::transport::MessageContext& context);
+    virtual void stop();
 
-    /* Common start and stop entrypoint
-     *
-     * Call the generic methods with the default global message context.
-     */
-    void start();
+    void setAsyncLatch();
 
-    virtual void stop();
+    void awaitAsyncLatch();
```

Comment:
These `setAsyncLatch` and `awaitAsyncLatch` functions are to support testing. Tests frequently check the side effects of sending an async message, then have to add some arbitrary sleep until waiting to check that it's done. This allows us to avoid the sleep and the inherent flakiness of that approach.